DOC-1671: Commit generated API docs (#2454)
tinymce/tinymce-docs: modules/ROOT/pages/apis/tinymce.addonmanager.adoc

:description: This class handles the loading of add-ons and their language packs. ThemeManager and PluginManager are instances of AddOnManager, and manage themes and plugins.
:keywords: add, get, load, requireLangPack
:moxie-type: api

This class handles the loading of add-ons and their language packs. ThemeManager and PluginManager are instances of AddOnManager, and manage themes and plugins.

[[summary]]
== Summary

[[methods-summary]]
=== Methods
[cols="2,5,1",options="header"]
|===
|Name|Summary|Defined by
|xref:#add[add()]|Adds a instance of the add-on by it's short name.|`xref:apis/tinymce.addonmanager.adoc[AddOnManager]`
|xref:#get[get()]|Returns the specified add on by the short name.|`xref:apis/tinymce.addonmanager.adoc[AddOnManager]`
|xref:#load[load()]|Loads an add-on from a specific url.|`xref:apis/tinymce.addonmanager.adoc[AddOnManager]`
|xref:#requireLangPack[requireLangPack()]|Loads a language pack for the specified add-on.|`xref:apis/tinymce.addonmanager.adoc[AddOnManager]`
|===

[[methods]]
== Methods

[[add]]
=== add()
[source, javascript]
----
add(id: String, addOn: tinymce.Theme | tinymce.Plugin): tinymce.Theme | tinymce.Plugin
----

> Adds a instance of the add-on by it's short name.

Reviewer: ```suggestion
Adds an instance of the add-on by its short name.
```
Typo corrections.

1. The indefinite article is *an* when the following sound is a vowel.
2. *Its* is the possessive pronoun. *It’s* is the abbreviation. FWIW, I remember this ‘exception’ by remembering that no possessive pronouns take an apostrophe (so: his, hers, theirs, its).

Reviewer 2: @brian-forte I think you missed an important part of the README updates here:
> **Note:** The API documentation should never be edited manually.

So as noted, this cannot be changed like this and any of these changes would need to be done on the TinyMCE source.

Reviewer 2: I'd suggest logging these changes in a Jira and getting them fixed that way instead.

==== Return value

* `xref:apis/tinymce.theme.adoc[Theme]` - The same theme or plugin instance that got passed in.
* `xref:apis/tinymce.plugin.adoc[Plugin]` - The same theme or plugin instance that got passed in.

'''

[[get]]
=== get()
[source, javascript]
----
get(name: String): tinymce.Theme | tinymce.Plugin
----
Returns the specified add on by the short name.

==== Parameters

* `name (String)` - Add-on to look for.

==== Return value

* `xref:apis/tinymce.theme.adoc[Theme]` - Theme or plugin add-on instance or undefined.
* `xref:apis/tinymce.plugin.adoc[Plugin]` - Theme or plugin add-on instance or undefined.

'''

[[load]]
=== load()
[source, javascript]
----
load(name: String, addOnUrl: String): Promise
----
Loads an add-on from a specific url.

==== Examples
[source, javascript]
----
// Loads a plugin from an external URL
tinymce.PluginManager.load('myplugin', '/some/dir/someplugin/plugin.js');

// Initialize TinyMCE
tinymce.init({
  ...
  plugins: '-myplugin' // Don't try to load it again
});
----

==== Parameters

* `name (String)` - Short name of the add-on that gets loaded.
* `addOnUrl (String)` - URL to the add-on that will get loaded.

==== Return value

> * `Promise` - A promise that will resolve when the add-on is loaded successfully or reject if it failed to load.

Reviewer: ```suggestion
* `Promise` — A promise that will resolve when the add-on loads successfully or reject if the addon fails to load.
```

1. Replaced the hyphen set open with an em-dash (also set open). The dash here is being used as a linking mark (it connects the word with its definition) and the em-dash is the grammatically accurate dash to use in this instance.
2. Edited the definition for tense. The original began in the simple present, switched to the simple past, then switched to the present participle. The edit is all in the simple present.
